game.py

```diff
@@ -20,7 +20,34 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    # Add a new list to track correctly guessed letters
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+
+    print()
+    print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES:
+        user_input = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+        
+        if user_input in correct_letter_guess_statuses:                        
+            print("You guessed a letter that's in the word!")
+            correct_letter_guess_statuses[user_input] = True
+        else:
+            print(f"The letter {user_input} is not in the word")
+            wrong_guesses_list.append(user_input)
+
+        print_snowman_graphic(len(wrong_guesses_list))
+        print(f"Wrong guesses: {wrong_guesses_list}")
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+
+        # for success
+        if is_word_guessed(snowman_word, correct_letter_guess_statuses):
+             print("Congratulations, you win!")
+             return
```

Comment:
Nice use of the `return` keyword. This allows for us to exit not only the while loop early if this condition is met but also the function as a whole. Something to consider here is readability, imagine if we decided to structure our loop like the following: 
```py
while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and is_word_guessed(snowman_word, correct_letter_guess_statuses): 
```
Which one more readily communicates to developers that this while loop is break condition is partially dependent on if the word is guessed? Why? (Note the checks themselves would function the same, though the following code of the modified `while` statement would look different in some places).